core/src/main/java/edu/wpi/grip/core/Palette.java

```diff
@@ -24,10 +29,54 @@
 
   @Subscribe
   public void onOperationAdded(OperationAddedEvent event) {
-    final OperationMetaData operation = event.getOperation();
-    map(operation.getDescription().name(), operation);
-    for (String alias : operation.getDescription().aliases()) {
-      map(alias, operation);
+    final OperationMetaData operationData = event.getOperation();
+    map(operationData.getDescription().name(), operationData);
+    for (String alias : operationData.getDescription().aliases()) {
+      map(alias, operationData);
+    }
+    // Validate that every input and output socket has a unique name and UID
+    Operation operation = operationData.getOperationSupplier().get();
+    try {
+      final List<? extends Socket> sockets = new ImmutableList.Builder<Socket>()
+          .addAll(operation.getInputSockets())
+          .addAll(operation.getOutputSockets())
+          .build();
+      checkDuplicates(
+          operationData,
+          "input socket names",
+          operation.getInputSockets(), s -> s.getSocketHint().getIdentifier()
+      );
+      checkDuplicates(
+          operationData,
+          "output socket names",
+          operation.getOutputSockets(), s -> s.getSocketHint().getIdentifier()
+      );
+      checkDuplicates(operationData, "socket IDs", sockets, Socket::getUid);
+    } finally {
+      operation.cleanUp();
+    }
```

Comment:
Not a huge fan of creating an operation when the user hasn't asked the UUID to create one.

Perhaps put this into a test that loads every operation?